File: backend/app/schemas/app_version.py
"""
Schemas for app version policy/update checks.
"""
from datetime import datetime
from typing import Literal, Optional

from pydantic import BaseModel


AppPlatform = Literal["ios", "android", "h5"]


class AppVersionPolicyResponse(BaseModel):
    platform: AppPlatform
    latest_version: str = ""
    latest_build: Optional[int] = None
    min_supported_version: str = ""
    min_supported_build: Optional[int] = None
    app_store_url: Optional[str] = None
    update_title: Optional[str] = None
    update_message: Optional[str] = None
    release_notes: Optional[str] = None
    is_enabled: bool = True
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class AppVersionPolicyUpdateRequest(BaseModel):
    platform: AppPlatform
    latest_version: str = ""
    latest_build: Optional[int] = None
    min_supported_version: str = ""
    min_supported_build: Optional[int] = None
    app_store_url: Optional[str] = None
    update_title: Optional[str] = None
    update_message: Optional[str] = None
    release_notes: Optional[str] = None
    is_enabled: bool = True


class AppVersionCheckResponse(BaseModel):
    platform: AppPlatform
    current_version: Optional[str] = None
    current_build: Optional[int] = None
    latest_version: str = ""
    latest_build: Optional[int] = None
    min_supported_version: str = ""
    min_supported_build: Optional[int] = None
    should_update: bool = False
    force_update: bool = False
    update_title: Optional[str] = None
    update_message: Optional[str] = None
    release_notes: Optional[str] = None
    app_store_url: Optional[str] = None
    checked_at: datetime

File: admin-dashboard/src/api/appVersion.ts
import { api } from './client';

export type AppPlatform = 'ios' | 'android' | 'h5';

export interface AppVersionPolicy {
  platform: AppPlatform;
  latest_version: string;
  latest_build: number | null;
  min_supported_version: string;
  min_supported_build: number | null;
  app_store_url: string | null;
  update_title: string | null;
  update_message: string | null;
  release_notes: string | null;
  is_enabled: boolean;
  created_at: string;
  updated_at: string;
}

export interface AppVersionPolicyUpdatePayload {
  platform: AppPlatform;
  latest_version: string;
  latest_build: number | null;
  min_supported_version: string;
  min_supported_build: number | null;
  app_store_url: string | null;
  update_title: string | null;
  update_message: string | null;
  release_notes: string | null;
  is_enabled: boolean;
}

export const getAdminAppVersionPolicy = async (platform: AppPlatform) => {
  const response = await api.get('/app-version/admin/policy', { params: { platform } });
  return response.data as AppVersionPolicy;
};

export const getAdminAppVersionPolicies = async () => {
  const response = await api.get('/app-version/admin/policies');
  return (response.data || []) as AppVersionPolicy[];
};

export const updateAdminAppVersionPolicy = async (payload: AppVersionPolicyUpdatePayload) => {
  const response = await api.put('/app-version/admin/policy', payload);
  return response.data as AppVersionPolicy;
};
